packages/astro/src/core/config.ts

```diff
@@ -483,6 +473,52 @@ export async function openConfig(configOptions: LoadConfigOptions): Promise<Open
 	};
 }
 
+interface TryLoadConfigResult {
+	value: Record<string, any>;
+	filePath?: string;
+}
+
+async function tryLoadConfig(configOptions: LoadConfigOptions, flags: CLIFlags, userConfigPath: string | undefined, root: string): Promise<TryLoadConfigResult | undefined> {
+	try {
+		// Automatically load config file using Proload
+		// If `userConfigPath` is `undefined`, Proload will search for `astro.config.[cm]?[jt]s`
+		const config = await load('astro', {
+			mustExist: !!userConfigPath,
+			cwd: root,
+			filePath: userConfigPath,
+		});
+
+		return config as TryLoadConfigResult;
+	} catch(e) {
+		if (e instanceof ProloadError && flags.config) {
```

Comment:
does this mean that `--config` won't work with `.ts` config files? Could this be moved to further down to be a final catchall, **_OR_** earlier in the function with a `fs.stat` call to check that the file exists before running proload or vite?

Reply:
No, this condition hits when the user passes a `--config` but that file doesn't exist. If the file does exist and is a `.ts` file it will either a) have worked through proload or b) caught here, but the `instanceof` check will fail and it will load through the subsequent Vite mechanism.